src/encoder.rs

```diff
@@ -960,25 +962,27 @@ impl<W: io::Write> UncompressedHeader for BitWriter<W, BigEndian> {
         self.write(12,0)?; // idc
         self.write(5, 31)?; // level
         self.write(1, 0)?; // tier
-        if seq.reduced_still_picture_hdr {
+        if fi.sequence.reduced_still_picture_hdr {
             unimplemented!();
         }
 
-        self.write_sequence_header(seq, fi)?;
+        self.write_sequence_header(fi)?;
 
-        self.write_color_config(seq)?;
+        self.write_color_config(&mut fi.sequence)?;
 
-        self.write_bit(seq.film_grain_params_present)?;
+        self.write_bit(fi.sequence.film_grain_params_present)?;
 
         self.write_bit(true)?; // trailing bit
 
         Ok(())
     }
 
-    fn write_sequence_header(&mut self, seq: &mut Sequence, fi: &FrameInvariants)
+    fn write_sequence_header(&mut self, fi: &mut FrameInvariants)
         -> io::Result<()> {
         self.write_frame_size(fi)?;
 
+        let mut seq = fi.sequence;
```

Comment:
You copy the sequence here (and modify the copy). Is it what you want?

Reply:
```suggestion
        let seq = &mut fi.sequence;
```

Reply:
:+1:

I've never used the GitHub "merge suggestion" feature, but it produces awful commits :scream: Maybe you could just squash/fixup it with the previous one :)